File: AgeCalculator/app/src/main/java/com/example/agecalculator/MainActivity.java
package com.example.agecalculator;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;


public class MainActivity extends AppCompatActivity {

   
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        EditText birthYear, newYear;
        Button findYear;
        TextView result;

        birthYear = findViewById(R.id.birthYear);
        newYear = findViewById(R.id.newYear);
        result = findViewById(R.id.result);
        findYear = findViewById(R.id.findYear);

        findYear.setOnClickListener(new View.OnClickListener () {
            @Override
            public void onClick(View view) {
                int birth = Integer.parseInt(birthYear.getText().toString());
                int year = Integer.parseInt(newYear.getText().toString());
                int total = year - birth;

                result.setText(""+total);
            }
        });



    }
}

File: DiceRoller_kt/app/src/main/java/com/example/diceroller/MainActivity.kt
package com.example.diceroller

import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val rollButton: Button = findViewById(R.id.rollDiceBtn)
        rollButton.setOnClickListener {
            val toast = Toast.makeText(this,"Dice Rolled!",Toast.LENGTH_SHORT)
            toast.show()
            rollDice()
        }


    }

    private fun rollDice() {
        val obj = Dice(6)
        val resultTextview : TextView = findViewById(R.id.rollDiceText)
        resultTextview.text = obj.roll().toString()
        obj.roll()

    }
}
class Dice(private val numSide : Int){
    fun roll(): Int{
        return (1..numSide).random()
    }
}